ligo/src/proposal.mligo

```diff
@@ -36,6 +36,82 @@ let ensure_proposal_is_unique (propose_params, store : propose_params * storage)
     then (failwith("PROPOSAL_NOT_UNIQUE") : proposal_key)
     else proposal_key
 
+// Gets the current period counting how many `voting_period` s have passed since
+// the 'started_on` timestamp. The periods start from zero index.
+let get_current_period(mvp_history, vp_length, sn : (vp_history option) * nat * timestamp) : nat =
+  let
+    (start_vp, start_ts) = match mvp_history with
+      | Some(lc) -> (lc.voting_period, lc.changed_on)
+      | None -> (0n, sn) in
+
+  match is_nat((Tezos.now - start_ts) : int) with
+        | Some (elapsed_time) -> start_vp + elapsed_time/vp_length
+        | None -> ([%Michelson ({| { FAILWITH } |} : string * unit -> nat)]
+            ("STARTED_ON_IN_FUTURE", ()))
```

Comment:
We can get rid of the `started_on` field in the storage as well as remove the `option` from the `vp_history`.

At origination we just need to set `vp_history` to `{voting_period = 0n; changed_on = now}`

Reply:
Good idea. Done.